packages/contracts/test/unirepVoting.test.ts

```diff
@@ -0,0 +1,209 @@
+//@ts-ignore
+import { ethers } from 'hardhat'
+import { deployUnirep, deployVerifierHelper } from '@unirep/contracts/deploy'
+import { SQLiteConnector } from 'anondb/node'
+import { schema, UserState } from '@unirep/core'
+
+import { Identity } from '@semaphore-protocol/identity'
+import { defaultProver } from '@unirep/circuits/provers/defaultProver'
+
+import {
+    genProofAndVerify,
+    genReputationCircuitInput,
+} from '@unirep/circuits/test/utils'
+import { Circuit, ReputationProof } from '@unirep/circuits'
+import { assert, expect } from 'chai'
+
+async function genUserStateInternal(id, app) {
+    // generate a user state
+    const db = await SQLiteConnector.create(schema, ':memory:')
+    const unirepAddress = await app.unirep()
+    const attesterId = BigInt(app.address)
+    const userState = new UserState({
+        db,
+        attesterId,
+        unirepAddress,
+        provider: ethers.provider,
+        id,
+        prover: defaultProver,
+    })
+    await userState.sync.start()
+    await userState.waitForSync()
+    return userState
+}
```

Comment:
I think it should not use `@unirep/core` in `@unirep/contracts`
because `@unirep/core` depends on `@unirep/contracts`

so we should bootstrap signup circuits, epochkey circuits, reputation circuits like other tests
another solution is that we put this test in `@unirep/core` and keep the vote contract here
so we can use `UserState` and its methods

Reply:
it seems like other tests involves user state lives in unirep/core makes sense that it belongs there. however do you know how getContractFactory works? after moving the contract cannot seem to be found by the test

Reply:
moved it to package core!